io.openems.edge.levl.controller/src/io/openems/edge/levl/controller/ControllerEssBalancingImpl.java

```diff
@@ -0,0 +1,593 @@
+package io.openems.edge.levl.controller;
+
+import java.time.Clock;
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
+import org.osgi.service.event.propertytypes.EventTopics;
+import org.osgi.service.metatype.annotations.Designate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+
+import io.openems.common.exceptions.OpenemsError.OpenemsNamedException;
+import io.openems.common.jsonrpc.base.JsonrpcRequest;
+import io.openems.common.jsonrpc.base.JsonrpcResponse;
+import io.openems.common.jsonrpc.base.JsonrpcResponseSuccess;
+import io.openems.edge.common.component.AbstractOpenemsComponent;
+import io.openems.edge.common.component.OpenemsComponent;
+import io.openems.edge.common.cycle.Cycle;
+import io.openems.edge.common.event.EdgeEventConstants;
+import io.openems.edge.common.jsonapi.Call;
+import io.openems.edge.common.jsonapi.ComponentJsonApi;
+import io.openems.edge.common.jsonapi.JsonApiBuilder;
+import io.openems.edge.controller.api.Controller;
+import io.openems.edge.ess.api.ManagedSymmetricEss;
+import io.openems.edge.ess.power.api.Phase;
+import io.openems.edge.ess.power.api.Pwr;
+import io.openems.edge.meter.api.ElectricityMeter;
+import io.openems.edge.levl.controller.common.Efficiency;
+
+import static java.lang.Math.round;
+import static java.lang.Math.min;
+import static java.lang.Math.max;
+
+@Designate(ocd = Config.class, factory = true)
+@Component(name = "Controller.Levl.Symmetric.Balancing", immediate = true, configurationPolicy = ConfigurationPolicy.REQUIRE)
+@EventTopics({ EdgeEventConstants.TOPIC_CYCLE_BEFORE_PROCESS_IMAGE, EdgeEventConstants.TOPIC_CYCLE_AFTER_WRITE, })
```

Comment:
CodeFormatting, See [FeneconBatteryExample](https://github.com/OpenEMS/openems/blob/d4d0df49c36a325941706d63dae50026842944b3/io.openems.edge.battery.fenecon.home/src/io/openems/edge/battery/fenecon/home/BatteryFeneconHomeImpl.java#L75-L84)

Reply:
maybe a stupid question, but if checkstyle does not complain code formatting should be ok, shouldn't it?

Reply:
@parapluplu You are right. If checkstyle does not complain, the code formatting should be ok. 

Regardless of this, using the code formatting like in [FeneconBatteryExample](https://github.com/OpenEMS/openems/blob/d4d0df49c36a325941706d63dae50026842944b3/io.openems.edge.battery.fenecon.home/src/io/openems/edge/battery/fenecon/home/BatteryFeneconHomeImpl.java#L75-L84) has some advantages. For example, for me it took 2-5s to understand the configuration of the fenecon battery. In your code it took me 15-20s. This is, because it took me some time to mentally divide the lines into several lines. This is how may brain is trained (because of other OpenEMS code). Other developers may have the same problem. 
As we all want to work efficiently on the same code base it would be nice if we format the code more or less identical. 
And in the end this issue should be reflected by an updated set of checkstyle rules..